run_test does not work on CKAN 2.3

When I try to run a harvester with the `run_test` paster command, I get the following error message:

```
(pyenv)vagrant@ord:/vagrant$ paster --plugin=ckanext-harvest harvester run_test 0f1a803a-a23e-4fef-aa18-60a56f166756 -c /home/vagrant/pyenv/src/ckan/development.ini
2016-01-11 15:43:04,664 DEBUG [ckanext.harvest.model] Harvest tables defined in memory
2016-01-11 15:43:04,672 DEBUG [ckanext.harvest.model] Harvest tables already exist

Running setup_app() from ckan.websetup
2016-01-11 15:43:05,229 DEBUG [ckanext.harvest.model] Harvest tables already exist
2016-01-11 15:43:05,390 INFO  [ckan.websetup] Creating tables: SUCCESS
Traceback (most recent call last):
  File "/home/vagrant/pyenv/bin/paster", line 9, in <module>
    load_entry_point('PasteScript==1.7.5', 'console_scripts', 'paster')()
  File "/home/vagrant/pyenv/local/lib/python2.7/site-packages/paste/script/command.py", line 104, in run
    invoke(command, command_name, options, args[1:])
  File "/home/vagrant/pyenv/local/lib/python2.7/site-packages/paste/script/command.py", line 143, in invoke
    exit_code = runner.run(args)
  File "/home/vagrant/pyenv/local/lib/python2.7/site-packages/paste/script/command.py", line 238, in run
    result = self.command()
  File "/vagrant/ckanext-harvest/ckanext/harvest/commands/harvester.py", line 163, in command
    self.run_test_harvest()
  File "/vagrant/ckanext-harvest/ckanext/harvest/commands/harvester.py", line 383, in run_test_harvest
    from ckanext.harvest.tests import lib
  File "/vagrant/ckanext-harvest/ckanext/harvest/tests/lib.py", line 1, in <module>
    from ckanext.harvest.tests.factories import HarvestSourceObj, HarvestJobObj
  File "/vagrant/ckanext-harvest/ckanext/harvest/tests/factories.py", line 4, in <module>
    from ckan.tests.factories import _get_action_user_name
  File "/home/vagrant/pyenv/local/lib/python2.7/site-packages/ckan-2.3.1b-py2.7.egg/ckan/tests/__init__.py", line 227, in <module>
    class WsgiAppCase(BaseCase):
  File "/home/vagrant/pyenv/local/lib/python2.7/site-packages/ckan-2.3.1b-py2.7.egg/ckan/tests/__init__.py", line 229, in WsgiAppCase
    assert wsgiapp, 'You need to run nose with --with-pylons'
AssertionError: You need to run nose with --with-pylons
```

Is there a simple fix for this?

--- a/ckanext/harvest/tests/factories.py
+++ b/ckanext/harvest/tests/factories.py
@@ -1,9 +1,9 @@
 import factory
 import ckanext.harvest.model as harvest_model
 try:
-    from ckan.tests.factories import _get_action_user_name
-except ImportError:
     from ckan.new_tests.factories import _get_action_user_name
+except ImportError:
+    from ckan.tests.factories import _get_action_user_name
 from ckan.plugins import toolkit
 
 

--- a/ckanext/harvest/tests/test_auth.py
+++ b/ckanext/harvest/tests/test_auth.py
@@ -7,9 +7,11 @@ from ckan.model import Package, Session
 from ckan.lib.helpers import url_for,json
 from ckan.lib.base import config
 
-
-from ckan.tests import CreateTestData
 #TODO: remove references to old tests
+try:
+    from ckan.tests import CreateTestData
+except ImportError:
+    from ckan.tests.legacy import CreateTestData
 try:
     from ckan.tests.functional.base import FunctionalTestCase
 except ImportError:

